### PR TITLE
grammar/style

### PR DESCRIPTION
**Pull request by real-or-random**
See individual commit messages

### CODE REVIEW

File: bip-musig2.mediawiki [hunk 1]
```diff
@@ -38,31 +38,31 @@ As a side effect, the number ''n'' of signers is not limited by any consensus ru
...
+* '''Key aggregation optionally independent of order''': The output of the key aggregation algorithm depends on the order of the input public keys. This standard defines a function to sort the public keys before key aggregation. This will ensure the same output, independent of the initial order. Key aggregation does not sort the public keys by default because applications often already have a canonical order of signers. Nonetheless, applications using this standar can mandate sorting before aggregation<ref>Applications that sort input public keys before aggregation should ensure that the sort implementation is reasonably efficient, and in particular does not degenerate to quadratic runtime on pathological inputs.</ref>.
```
Comment (jonasnick) [inline bip-musig2.mediawiki]: standar

File: bip-musig2.mediawiki [hunk 2]
```diff
@@ -94,16 +94,16 @@ Otherwise, it is possible to extract the secret signing key from the two partial
 To avoid accidental reuse of ''secnonce'', an implementation may securely erase the ''secnonce'' argument by overwriting it with 64 zero bytes after it has been read by ''Sign''.
 A ''secnonce'' consisting of only zero bytes is invalid for ''Sign'' and will cause it to fail.
 
-To simplify the specification, some intermediary values are unnecessarily recomputed from scratch, e.g., when executing ''GetSessionValues'' multiple times.
+To simplify the description, some intermediary values are unnecessarily recomputed from scratch, e.g., when executing ''GetSessionValues'' multiple times.
```
Comment (jonasnick) [inline bip-musig2.mediawiki]: Probably better "description of the algorithms".

File: bip-musig2.mediawiki [hunk 3]
```diff
@@ -75,14 +75,14 @@ Similarly, the test vectors that exercise the unimplemented features should be r
...
+Whenever they want to sign a message, the basic order of operations to create a multi-signature is as follows:
 
 '''First broadcast round:'''
 The signers start the signing session by running ''NonceGen'' to compute ''secnonce'' and ''pubnonce''<ref>We treat the ''secnonce'' and ''pubnonce'' as grammatically singular even though they include serializations of two scalars and two elliptic curve points, respectively. This treatment may be confusing for readers familiar with the MuSig2 paper. However, serialization is a technical detail that is irrelevant for users of MuSig2 interfaces.</ref>.
 Then, the signers broadcast their ''pubnonce'' to each other and run ''NonceAgg'' to compute an aggregate nonce.
 
 '''Second broadcast round:'''
-At this point, every signer has the required data to sign, which, in the specification, is stored in a data structure called [[#session-context|Session Context]].
+At this point, every signer has the required data to sign, which, in the description here, is stored in a data structure called [[#session-context|Session Context]].
```
Comment (jonasnick) [inline bip-musig2.mediawiki]: The word "here" is maybe confusing. It doesn't refer to the paragraph or section. Do you mean "description below"?

**Review by jonasnick — COMMENTED** [review]
Why do you prefer "standard" or "description of algorithms" over "specification"? I intentionally avoided the word "standard" because it's loaded (standard can mean that people adopted it).

**Comment by real-or-random** [comment]
> Why do you prefer "standard" or "description of algorithms" over "specification"?
>  I intentionally avoided the word "standard" because it's loaded (standard can mean that people adopted it).

Oh, I see, I think you had already mentioned that earlier to me, but I had forgotten. 

Okay, let's approach the terminology systemically. Here's my thinking when I created the PR:

There are two concepts we need to describe:
1. The "BIP MuSig2 way of doing multisigs". Candidates are "specification" and "standard", and "BIP". Before this PR, we use both "standard" (e.g., first two sentences *"This document proposes a standard for the protocol. The standard is compatible with [...])"*, and "specification" (e.g., *"The goal of this specification is to support a wide range of possible application scenarios."*), and I think we treat them as synonyms. There's also the word "BIP", which so far refers to the document itself (*"To help implementors understand updates to this BIP"*).
2. The "write-up = the text and the pseudocode". Candidates are "specification" and "description (of algorithms)". Note that we use the word "specification" also for this meaning (*"The MuSig2 variant in this specification stands out by combining all of the following features:"*). And I think this is even the more common meaning, i.e., a specification is a written description of a standard procedure, thing, whatever.

So currently, it's pretty inconsistent and can be confusing, which is my primary reason for opening this PR:
```
standard     specification
       \     /        \
        \   /          \
         way         write-up
```
Ideally, we'd have a bijection of concepts and terms, though I admit that sometimes it's hard to draw a border between the two concepts.

Now what this PR does:
1. Replace "specification" by "standard". This avoids using two words to denote the same concept. But, as you point out rightly, it has the disadvantage that some people may read standard as "well-established standard" instead of "thing which has been standardized".
2. Use "description". This gets rid of the overload of "specification". 

```
standard     description
       \              \
        \              \
         way         write-up
```


What about the following?
1. Use "BIP" (or even "proposal"). BIP means "proposal" after all, so it can totally refer to more than just the document, and it sounds natural in many places where we use "specification": *"If this BIP would nonetheless allow the final nonce to be the point at infinity, then the scheme would lose the following property"* . It's also very short (much less bulky than "specification"), and it doesn't have the "standard" problem. In one or two sentences, we need to rephrase a bit, or resort to a different term, but language is never perfect: *"This document proposes a BIP for the MuSig2 protocol." -- Not elegant. :) We may stick to "standard" here, and in this context it's clear that it's just a proposal.
2. Use "description", again because "specification" can be ambiguous.

```
    BIP     description
       \              \
        \              \
         way         write-up
```

**Comment by jonasnick** [comment]
> Use "description", again because "specification" can be ambiguous.

Are you suggesting just description or "description of the algorithms"? Maybe could alternatively say "specification of the algorithms" which is unambiguous and implies greater precision than just a "description".

> We may stick to "standard" here, and in this context it's clear that it's just a proposal.

I agree it's fine to say something like "propose a standard".

A minor nit is that I find "speicification" a bit more serious sounding than BIP (maybe because it overemphasizes Bitcoin?) but whatever.

**Comment by real-or-random** [comment]
> Are you suggesting just description or "description of the algorithms"? Maybe could alternatively say "specification of the algorithms" which is unambiguous and implies greater precision than just a "description"

What I had in mind was "description of the algorithms" when it's about the algorithms, and otherwise just say "description".

I'm happy to change this to "specification of the algorithms" when it's about the algorithms, and otherwise just say "specification" or "description", whatever feels right. I don't think we'll have many instances where it's not about the algorithms. (A quick look suggests that there are in fact none, so this sounds good to me.)

**Comment by jonasnick** [comment]
sgtm

**Comment by real-or-random** [comment]
I did this, and added some further refinements.

> A minor nit is that I find "speicification" a bit more serious sounding than BIP (maybe because it overemphasizes Bitcoin?) but whatever.

I agree, I then had the same feeling after I started using BIP everywhere. I mostly used "proposal" now, which looks natural to me at least. Sorry for changing plans agains.

**Review by jonasnick — COMMENTED** [review]
Nice,

ACK 15f131b7587f6fed539d9ad202e065713a7cca78

**Pull request merged**